langchain/src/callbacks/handlers/tracers.ts

```diff
@@ -114,14 +111,43 @@ export abstract class BaseTracer extends BaseCallbackHandler {
     this.runMap.set(run.uuid, run);
   }
 
-  protected async _endTrace(run: LLMRun | ChainRun | ToolRun): Promise<void> {
+  protected async _endTrace(run: Run): Promise<void> {
     if (!run.parent_uuid) {
       await this.persistRun(run);
-      this.executionOrder = 1;
     }
     this.runMap.delete(run.uuid);
   }
 
+  protected _getExecutionOrder(parentRunId: string | undefined): number {
+    // If a run has no parent then execution order is 1
```

Comment:
actually, this is not going to work because it looks at direct children, not at all children recursively.

let's say top level chain calls tool, then tool calls llm. Then the top level chain calls llm. The llm's execution order will incorrectly be computed as 3 (`parent_run.execution_order` = 1, number of direct children = 1)when it really should be 4